### PR TITLE
Issue 877/view column categorization

### LINKED ISSUE #877: View column categorization
## Description
A user should be able to find view column choices in the column modal based on their "topic" or "category", e.g. use case scenario or _type_ of data.

## Relevant Job Stories
* When I’m adding a column to a sheet, I want to be shown the full range of capabilities of view/sheet columns, so that I can discover new powerful ways of achieving what I want

## Prerequisites
Don't start this until #875 is done.

### Interface
- [x] The main area (right-hand side) of the column modal should group the choice cards by category
- [x] The left-hand side area of the column modal should show a list of categories
- [x] Clicking an option in the category list should scroll the main surface to that category

### Categorization
Exact categorization is TBD.

## Possible implementations
None

## Design specifications
image

## Open questions
What are the categories?

### PR DESCRIPTION
**Pull request by ziggabyte**
## Description
This PR adds a framework for adding categories to the column choice modal. Each category is represented in a menu on the left hand side, and clicking scrolls the corresponding category into view. 

Note: exact categories and what choices they contain is still TBD

## Screenshots
bild

## Changes
* Adds the `categories` file, where you can add categories and give them a list of choices.
* Adds a side menu to the `ColumnGallery` with names of all categories (invisible on mobile)

## Related issues
Resolves #877

### CODE REVIEW

File: src/features/views/components/ViewColumnDialog/categories.ts
Comment (richardolsson) [inline src/features/views/components/ViewColumnDialog/categories.ts:6]: Maybe this key should also be an enum?

File: src/features/views/components/ViewColumnDialog/ColumnGallery.tsx
Comment (richardolsson) [inline src/features/views/components/ViewColumnDialog/ColumnGallery.tsx:28]: The name `categoriesRef` is a bit misleading IMO, because it's used for the container that doesn't just contain a list of categories, but all of the choice cards. Do you agree?

Maybe something like `contentRef` or `choiceContainerRef` or something would be more clear?

**Review by richardolsson — DISMISSED** [review]
Great work! I found two minor things that I think could be improved, but they're not blockers. I'll let you decide what to do!

**Comment by rebecarubio** [comment]
Nicely done @ziggabyte 🚀 ! Good improvements with the enums in the choices file, I'll refactor my code to follow your pattern of names etc.

**Review by rebecarubio — DISMISSED** [review]

File: src/features/views/components/ViewColumnDialog/categories.ts
Comment (ziggabyte) [reply src/features/views/components/ViewColumnDialog/categories.ts:6]: agree! i think i thought about it and forgot

File: src/features/views/components/ViewColumnDialog/ColumnGallery.tsx
Comment (ziggabyte) [reply src/features/views/components/ViewColumnDialog/ColumnGallery.tsx:28]: i agree!

**Review by richardolsson — APPROVED** [review]
Nice! Ready to merge if tests pass. 💯

**Pull request merged**
